### PR TITLE
Color palette editor: add reset and save modal buttons, add telemetry

### PR DESCRIPTION
**Pull request by kimprice**
This PR adds reset and save buttons to the color palette editor.  Reset reverts the colors to the previously saved colors. 

I also added telemetry for when the color palette editor is opened and when users leave the palette editor with saved changes  (closes https://github.com/microsoft/pxt/issues/9268). 

I made a minor UI change to fix the spacing between the number and the color picker (closes https://github.com/microsoft/pxt/issues/9263).

Refer to https://github.com/microsoft/pxt/issues/9269 and https://github.com/microsoft/pxt-arcade/issues/5327

(Disregard commits from before Dec 13 - these were captured in my first color palette PR and already merged)

You can try it out here: https://arcade.makecode.com/app/4e350193157b215a7d487850a4d7c9b44c06ab16-f2ddffd761

Buttons are disabled when no changes have been made:
image
Buttons are enabled once a change is made:
image
Exit modal if closing without saving:
image

### CODE REVIEW

**Review by srietkerk — COMMENTED** [review]
The css/less variables in the only suggestion I have, everything else looks good!

File: theme/asset-editor.less [hunk 1]
```diff
@@ -242,4 +242,18 @@
...
     }
+}
+
+.palette-transparent-button.common-button {
+    background: none transparent;
+    border: none;
+    &.disabled {
+        background: none transparent;
+        border: none;
+        color: gray;
```
[the remark this answers is not in the document]
Comment (kimprice) [reply theme/asset-editor.less:253]: yea, I was wondering about this. Thanks!

File: theme/asset-editor.less [hunk 2]
```diff
@@ -242,4 +242,18 @@
     i.fas {
         transform: rotate(90deg);
     }
+}
+
+.palette-transparent-button.common-button {
+    background: none transparent;
```
Comment (jwunderl) [inline theme/asset-editor.less:248]: doesn't matter much but `background: none` is sufficient / clears out other all the other background-* sub properties for you so transparent shouldn't be needed~

```suggestion
    background: none;
```

**Review by jwunderl — APPROVED** [review]

File: theme/asset-editor.less (hunk 2 above)
Comment (kimprice) [reply theme/asset-editor.less]: thanks! I'll make that change in both spots.

**Pull request merged**